types/underscore/underscore-tests.ts

```diff
@@ -1726,6 +1742,13 @@ undefinedIdentityIterateeResult; // $ExpectType StringRecord
     extractChainTypes(_.chain(stringRecordList).drop(simpleNumber)); // $ExpectType ChainType<StringRecord[], StringRecord>
 }
 
+// compact
+{
+    _.compact(truthyFalsyList); // $ExpectType (string | number | true | object | Function | StringRecord | (() => void))[]
```

Comment:
I suppose I could make a type name for this to make this not super long, TS seems willing to simplify type names as much as it can wherever it can with types in the same namespace.

Reply:
Oh, ha, this doesn't already do this because `boolean` ends up being changed to `true` 😛

Reply:
Hmm, trying this out it doesn't seem to be willing to result in the simplified type and I'm not sure why 😕. It for sure always wants to simplify `StringRecord | undefined` to my custom `StringRecordOrUndefined` type (see e.g. assertions for `find`), which I've actually found to be more annoying than helpful.

Reply:
Oh, but you're referring to `''` and `0` being members of `string` and `number` that TS should track as excluded. Unfortunately I don't think TS can collapse or carry along negatives like that.

Reply:
So not really set-theoretic types after all. Shame.